server/src/test/java/org/opensearch/rest/RestControllerTests.java

```diff
@@ -623,6 +632,7 @@ public Exception getInboundException() {
         assertTrue(channel.getSendResponseCalled());
         assertThat(channel.getRestResponse().getHeaders().containsKey("Allow"), equalTo(true));
         assertThat(channel.getRestResponse().getHeaders().get("Allow"), hasItem(equalTo(RestRequest.Method.GET.toString())));
+        assertThat(channel.getRestResponse().content().utf8ToString(), containsString("Unexpected http method"));
```

Comment:
Looks like this test case shows the result of any unsupported HTTP method, since the handler for HTTP method will always throw an exception https://github.com/opensearch-project/OpenSearch/blob/2.0.0/server/src/test/java/org/opensearch/rest/RestControllerTests.java#L560.
I will add a separate test for it.

Reply:
@reta Sorry I find it very hard to simulate the real process after receiving an specific wrong HTTP method. The http method
The HTTP method that used to build [RestRequest](https://github.com/opensearch-project/OpenSearch/blob/2.0.0/server/src/main/java/org/opensearch/rest/RestRequest.java#L254) comes from interface [HttpRequest](https://github.com/opensearch-project/OpenSearch/blob/2.0.0/server/src/main/java/org/opensearch/http/HttpRequest.java#L68), and it has to be an item of the [enum](https://github.com/opensearch-project/OpenSearch/blob/2.0.0/server/src/main/java/org/opensearch/rest/RestRequest.java#L236). I couldn't find an existing class that implements interface `HttpRequest` can accept arbitrary HTTP method, and can be used in the test class. 
In real scenario, using any HTTP method other than those listed in the enum will get an exception, so I think the current code for an unit test is likely enough.

Reply:
Got it, sorry about that @tlfeng , thank you!